### ISSUE TITLE
Parse HTML snippets with html parser inside markdown

### ISSUE THREAD

**Issue by IgnusG** [textlint/textlint#665]
**What version of textlint are you using?**
11.6.3

**What file type (Markdown, plain text, etc.) are you using?**
markdown

**What did you do? Please include the actual source code causing the issue.**
```markdown
# This is a header

<details>
<summary>This should be parsed</summary>
So should this
</details>
```

**What did you expect to happen?**
The html is parsed (given a plugin for example) into separate nodes

**What actually happened? Please include the actual, raw output from textlint.**

The html is provided only as a whole chunk which makes it incompatible with a lot of plugins

![image](https://user-images.githubusercontent.com/6438760/77905835-3c893d80-7287-11ea-9e2c-017afb2e679b.png)

The current `textlint-html-plugin` doesn't work since it operates on html files. Adding markdown to the extensions lets the html parser take over completely which means normal markdown doesn't get parsed properly instead.

**Comment by azu** [comment]
Mixed parser is complex I think.
We have created [@textlint/textlint-plugin-markdown](https://github.com/textlint/textlint/tree/master/packages/%40textlint/textlint-plugin-markdown) as just markdown parser.
If the markwon parser parse html in markdown, it will make false-positive by some rule.
Also, rules need to support `summary` node type, `details` note type and more...

You can create `textlint-plugin-markdown-with-html`.
